src/installer/corehost/build.proj

```diff
@@ -33,6 +33,7 @@
       <BuildArgs Condition="'$(Compiler)' != ''">$(BuildArgs) $(Compiler)</BuildArgs>
       <BuildArgs Condition="'$(CMakeArgs)' != ''">$(BuildArgs) $(CMakeArgs)</BuildArgs>
       <BuildArgs Condition="'$(StripSymbols)' == 'true'">$(BuildArgs) -stripsymbols</BuildArgs>
+      <BuildArgs Condition="'$(TargetOS)' == 'FreeBSD'">$(BuildArgs) -os FreeBSD</BuildArgs>
```

Comment:
```suggestion
      <BuildArgs>$(BuildArgs) -os $(TargetOS)</BuildArgs>
```

Reply:
I used to do this but that breaks the builds. On Linux this will be changed to "Unix"  in Directory.Build.props and that breaks core host build. I don't know enough why we do that for installer but I'm afraid to change it. Or I can roll it back in the shell script and treat `Unix` as `Linux`.

Reply:
I also saw problems with "Unix as Linux" previously for FreeBSD and Android.

Re: https://github.com/dotnet/runtime/issues/33069 it would really mitigate the pain if RIDs are resolved transparently in just one place somewhere under `:/eng`, so we don't have to spent hours figuring out from MSBuild log dumps, just because of the "official" vs. "unofficial" distinctions too much spread all over the place (afaik, which is also a concept for only signing the package and can be confined to a single place to reduce unnecessary disparities and specializations, IMVHO).